lib/nostrum/api/base.ex

```diff
@@ -5,15 +5,27 @@ defmodule Nostrum.Api.Base do
 
   import Nostrum.Constants, only: [base_route: 0]
 
+  def request(conn, method, route, body, raw_headers, options) when is_list(options) do
```

Comment:
A question. Should we search-and-replace the current `options: []` with `options: %{}` to prevent building garbage here?

Reply:
I see that the `options` that gets passed to `HTTPoison` includes `params`, and that's the only key that Nostrum for options. So I'll change `options` to `params` wherever necessary and ignore the `:gun` specific `options`.